src/sentry/api/endpoints/organization_spans_tags.py

```diff
@@ -0,0 +1,149 @@
+from typing import cast
+
+import sentry_sdk
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features, options
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects, OrganizationEventsV2EndpointBase
+from sentry.api.paginator import SequencePaginator
+from sentry.api.serializers import serialize
+from sentry.api.utils import handle_query_errors
+from sentry.search.events.builder import SpansIndexedQueryBuilder
+from sentry.search.events.types import ParamsType, QueryBuilderConfig
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.tagstore.types import TagKey, TagValue
+
+
+@region_silo_endpoint
+class OrganizationSpansTagsEndpoint(OrganizationEventsV2EndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.PERFORMANCE
+
+    def get(self, request: Request, organization) -> Response:
+        if not features.has(
+            "organizations:performance-trace-explorer", organization, actor=request.user
+        ):
+            return Response(status=404)
+
+        try:
+            snuba_params, params = self.get_snuba_dataclass(request, organization)
+        except NoProjects:
+            return self.paginate(
+                request=request,
+                paginator=SequencePaginator([]),
+            )
+
+        max_span_tags = options.get("performance.spans-tags-key.max")
+
+        with handle_query_errors():
+            # This has the limitations that we cannot paginate and
+            # we do not provide any guarantees around which tag keys
+            # are returned if the total exceeds the limit.
+            builder = SpansIndexedQueryBuilder(
+                Dataset.SpansIndexed,
+                params=cast(ParamsType, params),
+                snuba_params=snuba_params,
+                query=None,
+                selected_columns=["array_join(tags.key)"],
+                # The orderby is intentionally `None` here as this query is much faster
+                # if we let Clickhouse decide which order to return the results in.
+                # This also means we cannot order by any columns or paginate.
+                orderby=None,
+                limitby=("array_join(tags.key)", 1),
+                limit=max_span_tags,
+                sample_rate=options.get("performance.spans-tags-key.sample-rate"),
```

Comment:
I don't think we'll ever need / want this sampled, we would prob want a materialized view instead in the long run or to control via period

Reply:
This is more of a stop gap solution to reduce the bytes scanned until we get there. Ive defaulted this to 1 so it doesn't do anything yet.